doc/console/Portable-Console-Device-High-Level-Design.md

```diff
@@ -0,0 +1,390 @@
+2. Only **one model** of console devices can work on SONiC switch at the same time. If console devices of different models are plugged in at the same time and the user doesn't manually specify `vendor_name` and `model_name`, none of these devices will work. (Look at [Factory Function Design](#factory-function-design) for more information.)
+3. Multiple console devices of the **same model** from the **same vendor** can be daisy-chained to extend more console ports. Whether daisy-chaining is supported and how many devices are supported in daisy-chain depends on the vendor's implementation.
+
+## Setup Portable Console Device in SONiC
+
+After the console device is plugged into the SONiC switch, we need to set up the console device in SONiC for further use. The setup process includes:
+
+  1. Install corresponding drivers.
+  2. Map console interface from `/dev/ttyUSB<id>` to `/dev/console-<line-number>`. (Usually done with the help of [udev](https://wiki.debian.org/udev))
+  3. Any other necessary steps.
+
+Vendors should package all the files and setup procedures into a corresponding `<vendor_name>-<model_name>.deb` file. Since we prefer the setup process completed after building a SONiC image, files used to build `.deb` packages should be put in the [Azure/sonic-buildimage](https://github.com/Azure/sonic-buildimage) repository, then the `.deb` packages will be built and installed automatically. 
+
+<!-- Pictures exported from Visio are zoomed to 70% -->
+
+![A flow diagram to explain the setup process](./Portable-Console-Device-High-Level-Design/setup.png)
+
+For example, vendor can prepare a `50-<vendor_name>-<model_name>.rules` file in the `.deb` package, and copy it to `/etc/udev/rules.d` during the installation phase. In the post installation phase, vendor can use `udevadm trigger -c add` to trigger the udev rules.
+
+### Udev `.rules` File Priority
+
+*This part proposes conventions for vendors who use [udev](https://wiki.debian.org/udev) to map console interface.*
+
+Since [`man udev`](https://man7.org/linux/man-pages/man7/udev.7.html) mentions that "all rules files are collectively sorted and processed in lexical order", the name of a udev `.rules` file usually start with a number (between 01 and 99) which specifies the priority. The larger the number, the higher the priority. If there is no special need, set the priority of the `.rules` file to 50. Then the file name will look like `50-<vendor_name>-<model_name>.rules`.
+
+## CONFIG_DB Changes
+
+### CONSOLE_SWITCH_TABLE
+
+The `CONSOLE_SWITCH_TABLE` holds the configuration database for the purpose of console switch features. This table is filled by the management framework. Three new fields will be added to this table.
+
+```
+; Console switch feature table
+key = CONSOLE_SWITCH:console_mgmt
+
+; field = value
+autodetect  = "enable"/"disable" ; "enabled" means factory function will auto detect which vendor's device is plugged in
```

Comment:
From Xin's comment:

We need add restrictions for these value. Such as, if autodetect is enabled, then vendor_name/model_name must be empty

Reply:
Do we need to mention it in this design doc? Or simply add restrictions when coding?

Reply:
Resolved in latest version.